### ISSUE TITLE
wasmldr: rework workload::run() and workload::Error

### ISSUE THREAD

**Issue by wgwoods** [enarx/enarx#997]
wasmldr's `workload` module was designed around the idea of a "workload" file format - a tarball containing a `.wasm` module and bundled configuration directives. We decided to drop the bundle format, so now `workload::run` just takes the `.wasm` file itself. This works OK, but we now have no way to configure `wasmtime` features or the WASI environment.

As the comments in wasmldr/src/workload.rs say:

> Since we're not bundling the launch/deployment config into `bytes`, the
> naive solution would just be to add new arguments for those things, like
> WasmFeatures, stdio handling, etc - but that gets messy quick.
> Instead we should probably refactor this into distinct steps, each with
> its own config options (and error variants - see above).

In short, we're going to need to rework the `workload` module before we can start adding things like proper stdio handling, TLS support, embedded file assets, etc.

This will probably look something like:

- [ ] Refactor to use the Builder pattern so we can add new configuration knobs as Builder functions
- [ ] Rework Error to make things align with the Builder (and satisfy clippy's complaints about the variant names)

**Comment by platten** [comment]
No longer applicable due to architecture changes.
